packages/cli-exec/src/exec.js

```diff
@@ -59,8 +59,12 @@ export const exec = command('exec', {
     log.warn('Percy is disabled');
   } else {
     try {
-      percy.projectType = percy.client.tokenType();
-      percy.skipDiscovery = percy.shouldSkipAssetDiscovery(percy.projectType);
+      // Skip this for app because they are triggered as app:exec
+      // Remove this once they move to exec command as well
+      if (percy.projectType !== 'app') {
```

Comment:
Also this is not correct way to handle this, we need to throw error and exit from here.

Ideally do at top of the function, and then run the usual flow.

Reply:
Why we want to throw error?, if its `app:exec` the current flow should run as it is and no override should take place. Else override the projectType which we were doing before on the basis of tokenType.

Reply:
Understood, lets update test to reflect this cross usage behaviour.

app:exec with old and new token

exec with old and new app token

Reply:
@itsjwala  I tried `log.debug` however we don't log `log.debug` in any category neither in `stdout` nor in `stderr`. Hence I didn't have anything to match against expectation.